lib/phoenix_live_view/channel.ex

```diff
@@ -846,7 +859,9 @@ defmodule Phoenix.LiveView.Channel do
     # ever is a LiveView connection, the view won't be loaded and
     # the mount/handle_params callbacks won't be invoked as they
     # are optional, leading to errors.
-    view.__live__()
+    %{at_mount: at_mount} = view.__live__()
+
+    lifecycle = Lifecycle.__attach_at_mount__(%Lifecycle{}, at_mount)
```

Comment:
The Lifecycle module should be private, so theoretically we don't need underscores in this function. Perhaps we should also move the struct building in there?

```suggestion
    lifecycle = Lifecycle.mount(at_mount)
```

Reply:
Actually, we can probably build the lifecycle struct at compilation time for `__live__`. This way we don't need to do this work on mount every time.

Reply:
Done!